pipeline/scripts/13-excelExport.R

```diff
@@ -398,39 +398,49 @@ ggsave(paste0(outdir, "/plots/IS_line_select_sum.png"), plot = IS_sum_selection_
 
 ### POSITIVE CONTROLS CHECK
 # these positive controls need to be in the samplesheet, in order to make the Pos_Contr.RData file
-positivecontrol_list <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')
+# Positive control samples all have the format P1002.x, P1003.x and P1005.x (where x is a number)
+
+# positivecontrol_list <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')
+column_list <- colnames(outlist)
+patterns <- c("^(P1002.)[[:digit:]]+_", "^(P1003.)[[:digit:]]+_", "^(P1005.)[[:digit:]]+_")
+positive_controls_index <- grepl(pattern=paste(patterns, collapse="|"), column_list)
+positivecontrol_list <- column_list[positive_controls_index]
+
 if (z_score == 1) {
-  # find if one or more positive control samples are missing and put in list
+  # find if one or more positive control samples are missing
   missing_pos <- c()
-  for (pos in positivecontrol_list) {
-    if (!(pos %in% colnames(outlist))) {
-      missing_pos <- c(missing_pos, pos)
-    }}
+  if (length(positivecontrol_list) != 3){
+    missing_pos <- paste0(c("positive controls list is not complete: ", positivecontrol_list), collapse=" ")
+  }
   # you need all positive control samples, thus starting the script only if all are available
```

Comment:
I think the outcome of this positive control check became a bit different and I am not sure whether this was intended. Before the update if one of the three required controls was missing the code would always give an error. After the update the code will give an error when the number of control is not 3.

Some examples:

`c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')` Before: no error, After: no error.
`c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore', 'P1005.1_Zscore')` Before: no error, After: error.
`c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1003.1_Zscore')` Before: error, After: no error.

Reply:
those three specifically always need to be there (hardcoded). The ones that were not inside the colnames would end up in "missing_pos" and mentioned in the end mail.

because it was hardcoded before, but now I get the positive controls from the colnames itself and then would check again using the colnames (which does nothing) so I changed it to be 3
It was always the case, that if not all three were present, it would not run the code below. The only thing that will change is that in de end mail it will only tell that some are missing, not specifically which ones.

Reply:
After communicating via Teams chat (if one/multiple are present multiple times, or not all three specifically are present, the current code fails.)

used the following code to add this functionality
first 2 should give a go! (all three specific ones are present, even though double)
last 2 should give a no! (not all three specific ones are present = P1002, P1003 and P1005)

```
positive_controls <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore')
positive_controls <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1005.1_Zscore', 'P1005.1_Zscore')
positive_controls <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1006.1_Zscore')
positive_controls <- c('P1002.1_Zscore', 'P1003.1_Zscore', 'P1003.1_Zscore')
if (any(grep("^(P1002\\.)[[:digit:]]+_", positive_controls)) && any(grep("^(P1003\\.)[[:digit:]]+_", positive_controls)) && any(grep("^(P1005\\.)[[:digit:]]+_", positive_controls))){
  print("all three are present")
} else {
    print("nope")
  }
```

Reply:
Looks good, maybe for readability place each test on a separate line:
```R
if (
  any(grep("^(P1002\\.)[[:digit:]]+_", positive_controls)) && 
  any(grep("^(P1003\\.)[[:digit:]]+_", positive_controls)) && 
  any(grep("^(P1005\\.)[[:digit:]]+_", positive_controls))
){
  print("all three are present")
} else {
  print("nope")
}
```